mrjob/fs/gcs.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import fnmatch
+import logging
+
+# from mrjob.aws import s3_endpoint_for_region
+from mrjob.fs.base import Filesystem
+from mrjob.parse import urlparse
+from mrjob.runner import GLOB_RE
+
+
+from apiclient import discovery
+from oauth2client.client import GoogleCredentials
+from apiclient.http import MediaIoBaseDownload
+from apiclient import http
+import io
+import json
+import base64
+import binascii
+
+log = logging.getLogger(__name__)
+
+
+class GCSFilesystem(Filesystem):
+    """Filesystem for Google Cloud Storage (GCS) URIs. Typically you will get one of these via
+    ``DataprocJobRunner().fs``, composed with
+    :py:class:`~mrjob.fs.ssh.SSHFilesystem` and
+    :py:class:`~mrjob.fs.local.LocalFilesystem`.
+    """
+
+    def __init__(self):
+        credentials = GoogleCredentials.get_application_default()
+        self.service = discovery.build('storage', 'v1', credentials=credentials)
```

Comment:
Probably should call this `self._service` unless you're confident you want to expose it with this name.